### PR TITLE
✨ Yellow story ads progress bar for ads

### PR DESCRIPTION
**Pull request by powerivq**
It is gated under the story-ad-auto-progression experiment. Can you do a first round of review? Tests to be added.

image

### CODE REVIEW

File: extensions/amp-story/1.0/progress-bar.js [hunk 1]
```diff
@@ -417,58 +422,21 @@ export class ProgressBar {
...
+   * @param {boolean} isAd
    * @private
    */
-  buildSegmentEl_() {
+  buildSegmentEl_(isAd) {
     const segmentProgressBar = (
       <li class="i-amphtml-story-page-progress-bar">
-        <div class="i-amphtml-story-page-progress-value"></div>
+        <div
+          class={
```
Comment (mszylkowski) [inline extensions/amp-story/1.0/progress-bar.js:435]:
```js
class={objStr({
        'class-1': true,
        'class-2': isAd,
      })}
```

**Review by mszylkowski — COMMENTED** [review]
Looks good so far

File: extensions/amp-story/1.0/progress-bar.js [hunk 2]
```diff
@@ -218,6 +213,16 @@ export class ProgressBar {
     return root;
   }
 
+  /**
+   * Whether the given id corresponds to an ad page.
+   * @param {string} id
+   * @return {boolean}
+   * @private
+   */
+  isAdPage_(id) {
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/progress-bar.js:222]: Sorry about this, I'll keep being picky about how we name things, I believe this is super important for our ever growing team and writing easy to maintain code. We're talking about segments in this file, not pages

```suggestion
  isAdSegment_(id) {
```

File: extensions/amp-story/1.0/test/test-amp-story-progress-bar.js [hunk 3]
```diff
@@ -30,20 +31,26 @@ describes.realWin('amp-story-progress-bar', {amp: true}, (env) => {
...
     ]);
-    const progressBar = ProgressBar.create(env.win, storyEl).build('page-1');
-    doc.body.appendChild(progressBar);
+    progressBar = ProgressBar.create(env.win, storyEl);
+    doc.body.appendChild(progressBar.build('page-1'));
   });
 
   describe('story ad progress segment', async () => {
-    it('should create/remove ad segment based on ad visibility', () => {
+    it('should create progress bar with ad pages', () => {
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/test/test-amp-story-progress-bar.js:39]: Please add another test that checks that there's the correct count of segments in the progress bar

**Review by gmajoulet — APPROVED** [review]

File: extensions/amp-story/1.0/progress-bar.js (hunk 1 above)
Comment (powerivq) [reply extensions/amp-story/1.0/progress-bar.js]: done

File: extensions/amp-story/1.0/test/test-amp-story-progress-bar.js (hunk 3 above)
Comment (powerivq) [reply extensions/amp-story/1.0/test/test-amp-story-progress-bar.js:47]: Done

**Pull request merged**